Adds in Lichdom and Lesser Summon Guns Spells--Wiz Tweaks

Fox-McCloud opened the pull request:
Ports of TG's Lichdom and Lesser Summon Guns Spells for Wizards (https://github.com/tgstation/-tg-station/pull/9879 and https://github.com/tgstation/-tg-station/pull/14146) in addition to making some wizard spell/item availability tweaks from TG as well.

Lichdom is a single use wizard spell that does about what you expect it to: turns you into an immortal lich.

When the spell is used it will bind your soul to a phylactery---that being whatever item is in your hands---when said item is turned into a phylactery it will take on a green color and have a unique description to it.

Once you've done this, you will become a lich in full--your previous species, whatever it was, is now gone; you'll be a skeleton. 

The biggest benefit of this is immortality. How does this work?

When you die, you can active this spell again and you'll instantly revive at the location of your phylactery and become weakened for a s time---there will be a distinct green trail that will point to where you respawned at. Your old body will drop all of its belongings and dust (so it's not without consequence).

Each time you die, the cooldown of the spell increased, in addition to how long you're weakened when you revive at your phylactery's location


So, how do you kill a lich?

Simple: destroy their brain---either rip it out and destroy it or put the lich in the creamatorium/gibber; anything else, and he can revive through it (this includes gibbing, as long as the brain isn't deleted in the process of things).

Option two? Find the phylactery and destroy it; the Lich will then stay dead--that said, for the round to properly end, you'll still want to delete his brain.

Lesser Summon Guns
 - It's an expensive spell that generates 30 rifles in your hand that you can fire; it costs 4 points to purchase (out of 10)

Spell Book Tweaks
- Overall spell book points has been increased from 5 to 10
- Default spell cost has been increased from 1 to 2
- The follower spells have a lower cost now:
 - disable tech
 - smoke
 - force wall
 - knock
 - summon item
 - Staff of Door Creation

Spell Book Additions/Tweaks
- Adds in the charge spell
 - Can be used to recharge one use spell books, wands, and even cells
- Adds in Wand of Healing
- Adds in Mjolnir to purchase
- Adds in the singularity hammer to purchase
- Reorganizes and re-arranges the spell book to be easier to find items/spells

Other Tweaks
- Magic Missile has a longer cooldown and no longer deals damage, but has a lower cooldown when upgraded
- Knock now opens+unlocks lockers
- Mjolnir no longer does an extra 30 burn damage on hit, but it does have a throw impact now that stuns the person hit
- Mjolnir stun shortened from 10 to 3

Fixes
- Fixes being able to upgrade summon item multiple times
- Fixes spells not being properly removed/deleted
- Fixes HUDs not being properly removed when carbons are destroyed
- Fixes the charge spell so it actually works on other mobs mind's as opposed to just their mob spells
- Fixes Blind spell making the target nearsighted instead of actually blind



:cl: Fox McCloud
rscadd: Adds in lichdom spell for the wizard
rscadd: Adds in lesser summon guns spell for wizard
rscadd: adds in Mjolnir and the singularity hammer as weapons the wizard can purchase
rscadd: Wizard can purchase the charge spell from the spell book
rscadd: Wizard can purchase a healing staff from the spell book
tweak: Tweaks and lowers spell costs of utility spells for wizard
tweak: Re-organizes the spellbook to be better divided into categories and have a better window size
tweak: Knock now opens+unlocks lockers
tweak: Magic Missile has a slightly longer cooldown and no longer deals damage, but has a lower cooldown is upgraded
/:cl:

Crazylemon64 commented on code/datums/spells/lichdom.dm:
```diff
@@ -0,0 +1,129 @@
+/obj/effect/proc_holder/spell/targeted/lichdom/New()
+	if(!config.continous_rounds)
+		existence_stops_round_end = 1
+		config.continous_rounds = 1
```
This... seems like a bizarre solution... We might just want to have a function "Is out of commision" for our "are all antags dead" check, and make "having this spell" have them still count as in commision, even if they're "dead'

Fox-McCloud replied:
TG has something similar to what you describe, but I don't have the time, energy, or want-to to refactor gamemode code and port over Mulligan, so this is the nearest equal that doesn't require a lot of changes to function.

Crazylemon64 replied:
As for "making the round properly end", you can have the spell check if it can revive the person, with another proc to check if the phylactery exists

Crazylemon64 replied:
I'll see about getting something more elegant going, once this is merged - I think this might cause more than a few mutual-exclusion problems in situations with multiple liches.

Fox-McCloud replied:
the spell ti revive is not automatic; you have to cast it yourself; the way the spell is designed is that the game mode absolutely will not end until the wiznerd's brain is totally decimated.

Crazylemon64 commented:
Hmmm... I can't complain about this, but this WOULD put a damper for me to have milk as an advanced healing item for skeletons, since they can show up and have definite agency now... Unless you think that would still be cool?

Crazylemon64 commented on code/datums/spells/summonitem.dm:
```diff
@@ -7,7 +7,7 @@
 	invocation_type = "whisper"
 	range = -1
-	level_max = 1 //cannot be improved
+	level_max = 0 //cannot be improved
```
Ah, the reasoning for this change...? What exactly will this do?

Fox-McCloud replied:
Because you're not supposed to be able to upgrade it.

Crazylemon64 commented:
What exactly prevents the wizard from being a twerp and taking robeless casting and "area teleport", and then using this on his spawn area to become effectively unstoppable?

Fox-McCloud commented:
>What exactly prevents the wizard from being a twerp and taking robeless casting and "area teleport", and then using this on his spawn area to become effectively unstoppable?

You have to cast lichdom on the station z-level or else it will fail, to prevent just the thing you describe.

Crazylemon64 commented on code/datums/spells/lichdom.dm:
```diff
@@ -0,0 +1,129 @@
+	school = "necromancy"
+	charge_max = 10
+	clothes_req = 0
+	centcom_cancast = 0
```
Ah, this line prevents the "being a twerp" clause, yes?

Fox-McCloud replied:
As does this line 60

Crazylemon64 commented on code/datums/spells/lichdom.dm:
```diff
@@ -0,0 +1,129 @@
+			lich << "<span class='warning'>Your bones clatter and shutter as they're pulled back into this world!</span>"
+			charge_max += 600
+			var/mob/old_body = current_body
+			var/turf/body_turf = get_turf(old_body)
```
How does get_turf behave when passed null?

Fox-McCloud replied:
It doesn't generate a runtime, if that's what you're getting at.

Tastyfish replied:
It returns null.

Earthdivine commented:
Finnaly I can stop adminboosing to make liches.

NTSAM commented:
I like this alot.
:+1:

Tastyfish commented:
I could see an interesting suicide attack play-style come out of this. :+1:

Fox-McCloud commented:
Was going to add some of these things in another PR, but I figured...eh, might as well; there's bee complaints about wizards win-rates anyway

Added in the lesser summon guns spell and added in a number of new things the wizman can buy+tweaked their cost. Please see OP.

Crazylemon64 commented:
> Guns, Timestop, and a guardian

Dio wizard when

Glorken commented on code/datums/spells/lichdom.dm:
```diff
@@ -0,0 +1,130 @@
+			lich.real_name = M.mind.name
+			M.mind.transfer_to(lich)
+			lich.set_species("Skeleton")
+			lich << "<span class='warning'>Your bones clatter and shutter as they're pulled back into this world!</span>"
```
Should this be shudder instead of shutter?

Regen1 commented:
>It's an expensive spell that generates 30 rifles in your hand that you can fire; it costs 4 points to purchase (out of 10)

I'm sorry, but did I misunderstand? You summon 30 rifles into your hand that all fire?

>Adds in Mjolnir to purchase
Adds in the singularity hammer to purchase

Not a supporter of this, it ruins the whole "Wizard spellcaster" thing, melee weapons on wizards aren't really great, and it will force you to either get the Recall Item spell as well, or risk getting disarm-pushed 360 windmill slammed by the first assistant.

Crazylemon64 commented:
> I'm sorry, but did I misunderstand? You summon 30 rifles into your hand that all fire?

It summons a gun, that when fired, you automatically drop, and grab another one.

Crazylemon64 commented:
You should probably double the reward values of the challenges too, so they don't get too small to be useful

Regen1 commented:
ohhhhh
thanks :P

Fox-McCloud commented:
Alright, updated a few more spells:

- Knock now opens+unlocks lockers
- Magic missile has a longer CD, overall, and doesn't do damage now, but has a lower CD if upgraded
- improves the behavior of "charge" spell

We're pretty much at complete parity, with TG, with wizard, once this PR gets merged---the only exception is the lack of the shapeshift spell (which I don't feel like making the necessary changes to, yet to make that work without any issues).

Crazylemon64 commented:
What about the supermatter sword \</joke\>
